### ISSUE TITLE
Unable To Access Rear Camera on iPad iOS 11.4.1

### ISSUE THREAD

**Issue by bm2112** [schmich/instascan#172]
When I launch the website it will show that is displaying the rear camera, but it is actually displaying the front. When I switch over to the front camera, it also displays the front. I've tried everything I could find on the internet to fix this issue and I've had no luck. I've tried making the facingMode "environment", using the code from a couple of forks of this project which supposedly allow the iOS rear camera to work, and changing large portions of the app.js and camera.js myself. I've manually selected camera[1] instead of camera[0] and that will cause it to launch with the other camera selected, but it still displays the front camera.

Has anyone else had an issue with the rear camera on iOS that was unable to solve it using one of the above methods I mentioned? I would greatly appreciate some help. The website works fine for Android and Windows for many browsers, but iOS Safari is giving me nothing but trouble. Thanks

**Comment by augustosamame** [comment]
As indicated in the Readme, this library only works for mobile browsers that implement WebRTC. IOS browsers currently do not. 

> Instascan works on non-iOS platforms in any browser that supports the WebRTC/getUserMedia API, which currently includes Chome, Firefox, Opera, and Edge. IE and Safari are not supported.

**Comment by bm2112** [comment]
That is not true. Safari in the newest version of iOS  supports WebRTC API. As I said above, it works for the front camera, but not the rear camera.

**Comment by jvanalst** [comment]
@bm2112 is correct. Much of WebRTC was made available in iOS and Safari 11. Certain things like MediaRecorder are still missing, but it looks like everything used in this library (MediaStreams) should be available.

Maybe the issue should be updated to say "add iOS 11 support" since it's most of the way there anyway.

**Comment by kibagami-jubei** [comment]
Has anyone resolved this? I

**Comment by kibagami-jubei** [comment]
Yeah, been working on this, I can't get this to work. Trying to go back and re-read threads. I got the front-facing camera to work by adding the latest web RTC adapter and adding the playsinline attribute to the video tag. It can read the cameras but does nothing when swapping to the back camera.

**Comment by guidomgs** [comment]
I m not sure how to change this, but when you start the camera is posible to change the camera


 var constraints = {
            video: {
                facingMode: "environment" //"user" for selfie `mode`
            },
            audio: false
        };

function cameraStart() {
            navigator.mediaDevices
                .getUserMedia(constraints)
                .then(function(stream) {
                    track = stream.getTracks()[0];
                    cameraView.srcObject = stream;
                })
                .catch(function(error) {
                    console.error("Oops. Something is broken.", error);
                });
        }
